cmd/create/testoperatorrelease/requests.go

```diff
@@ -0,0 +1,226 @@
+package testoperatorrelease
+
+import (
+	"context"
+	"fmt"
+	"io/ioutil"
+	"path/filepath"
+	"regexp"
+	"strings"
+
+	"github.com/Masterminds/semver/v3"
+	"sigs.k8s.io/yaml"
+
+	"github.com/giantswarm/apiextensions/v2/pkg/apis/release/v1alpha1"
+	"github.com/giantswarm/microerror"
+)
+
+type apprequest struct {
+	Name    string `json:"name"`
+	Version string `json:"version"`
+	Issue   string `json:"issue"`
+}
+
+type request struct {
+	Name     string       `json:"name"`
+	Requests []apprequest `json:"requests"`
+}
+
+type requests struct {
+	Releases []request `json:"releases"`
+}
+
+func (r *runner) updateFromRequests(ctx context.Context, release *v1alpha1.Release) error {
+	reqs := requests{}
+	// Parse requests from the requests.yaml file.
+	{
+		requestsPath := filepath.Join(r.flag.ReleasesPath, r.flag.Provider, "requests.yaml")
+
+		releaseYAML, err := ioutil.ReadFile(requestsPath)
+		if err != nil {
+			return microerror.Mask(err)
+		}
+
+		err = yaml.Unmarshal(releaseYAML, &reqs)
+		if err != nil {
+			return microerror.Mask(err)
+		}
+	}
+
+	targetVersion, err := semver.NewVersion(strings.TrimPrefix(release.Name, "v"))
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	r.logger.Debugf(ctx, "Looking for requests for version %v", targetVersion.String())
+
+	apps, err := r.mergeRequirements(ctx, reqs, *targetVersion)
+	if err != nil {
+		return microerror.Mask(err)
+	}
+
+	for appname, minVersionStr := range apps {
+		// Search app or component in the release CR.
+		found := false
+		for i, comp := range release.Spec.Components {
+			if comp.Name == appname {
+				found = true
+				currentVersion, err := semver.NewVersion(comp.Version)
+				if err != nil {
+					return microerror.Mask(err)
+				}
+
+				minVersion, err := semver.NewVersion(minVersionStr)
+				if err != nil {
+					return microerror.Mask(err)
+				}
+
+				if currentVersion.LessThan(minVersion) {
+					release.Spec.Components[i].Version = minVersion.String()
+				}
+				break
+			}
+		}
+
+		if found {
+			continue
+		}
+
+		for i, app := range release.Spec.Apps {
+			if app.Name == appname {
+				found = true
+				currentVersion, err := semver.NewVersion(app.Version)
+				if err != nil {
+					return microerror.Mask(err)
+				}
+
+				minVersion, err := semver.NewVersion(minVersionStr)
+				if err != nil {
+					return microerror.Mask(err)
+				}
+
+				if currentVersion.LessThan(minVersion) {
+					release.Spec.Apps[i].Version = minVersion.String()
+				}
+				break
+			}
+		}
+
+		if !found {
+			r.logger.LogCtx(ctx, "level", "warning", "message", fmt.Sprintf("Adding a new component from the requirements is not currently supported. Tried to add %s", appname))
```

Comment:
Not sure if I understand this error message. This will happen when there is an `app` in the `requests` file that it's not in the `Release`?

Reply:
@fiunchinho yes. If we add a new app or component in the request that is not present in the latest release (like `cluster-autoscaler` in 13.1.0) there is no easy way to know if that is an `app` or a `component` so we don't know where to put it. That's the reason why this is an error.
I didn't want this case to be blocking, hence the decision to make it a warning and continue with the next apps.
WDYT?

Reply:
Maybe the error message could be like:
`"found unknown component %q from requests.yaml while it's not present in latest release.yaml; skipping", appname` ?

Reply:
agreed

Reply:
perfect